solara/components/file_drop.vue

```diff
@@ -16,16 +23,17 @@ module.exports = {
       event.preventDefault();
       const items = await Promise.all([...event.dataTransfer.items]);
       const files = items.map(i => i.webkitGetAsEntry())
-      const fileHolder = files.filter(f => f.isFile)[0]
-      const file = await new Promise((rs, rj) => fileHolder.file(rs, rj))
+      const fileHolders = files.filter(f => f.isFile)
+      const nativeFilesPromises = fileHolders.map(fileHolder => new Promise((rs, rj) => fileHolder.file(rs, rj)))
+      const nativeFiles = await Promise.all(nativeFilesPromises)
 
-      this.native_file_info = [file]
+      this.native_file_info = nativeFiles
       this.file_info = this.native_file_info.map(
-          ({name, isFile, size}) => ({
-            name,
-            isFile,
-            size,
-          }));
+        ({name, isFile, size}) => ({
+          name,
+          isFile: !!isFile, // Ensure this is a boolean
```

Comment:
Why is this needed? The spec says it's a boolean: https://developer.mozilla.org/en-US/docs/Web/API/FileSystemEntry/isFile

Reply:
Yes `FileSystemEntry` has `isFile` property. However,  after applying Promises (which effectively creates File objects), we don't have isFile property anymore. So isFile of the returned object was undefined. This was true for the original version. By using !!isFile, I was trying to create a boolean variable from undefined. Anyway, it turns out that isFile is not important because it is never used after this place. So this bug was not causing any problem. 

I'm suggesting the following. I think it is ok to set isFile=true because in the beginning we already filter files.

~~~javascript
this.file_info = this.native_file_info.map(
  ({name, size}) => ({
    name,
    isFile: true,
    size,
  }));
~~~